source/DSCResources/MSFT_SccmIniFIle/MSFT_SccmIniFIle.psm1

```diff
@@ -0,0 +1,1107 @@
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SMSInstallDir,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SDKServer,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $PreRequisiteComp,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $PreRequisitePath,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $AdminConsole,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $JoinCeip,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $MobileDeviceLanguage,
+
+        [Parameter()]
+        [ValidateSet('EnforceHTTPS','HTTPorHTTPS')]
+        [String]
+        $RoleCommunicationProtocol,
+
+        [Parameter()]
+        [Boolean]
+        $ClientsUsePKICertificate,
+
+        [Parameter()]
+        [String]
+        $ManagementPoint,
+
+        [Parameter()]
+        [ValidateSet('HTTPS','HTTP')]
+        [String]
+        $ManagementPointProtocol,
+
+        [Parameter()]
+        [String]
+        $DistributionPoint,
+
+        [Parameter()]
+        [ValidateSet('HTTPS','HTTP')]
+        [String]
+        $DistributionPointProtocol,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $AddServerLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $AddClientLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $DeleteServerLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $DeleteClientLanguages,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SQLServerName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $DatabaseName,
+
+        [Parameter()]
+        [String]
+        $SqlSsbPort,
+
+        [Parameter()]
+        [String]
+        $SQLDataFilePath,
+
+        [Parameter()]
+        [String]
+        $SQLLogFilePath,
+
+        [Parameter()]
+        [String]
+        $CCARSiteServer,
+
+        [Parameter()]
+        [String]
+        $CasRetryInterval,
+
+        [Parameter()]
+        [ValidateRange(0, 100)]
+        [uint16]
+        $WaitForCasTimeout,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $CloudConnector,
+
+        [Parameter()]
+        [String]
+        $CloudConnectorServer,
+
+        [Parameter()]
+        [Boolean]
```

Comment:
I believe this could have a validateSet I think the only options available for UseProxy is 0  and 1

Reply:
Use proxy is already a boolean?